site/src/pages/docs/client-decorator.mdx

```diff
@@ -169,6 +169,77 @@ preview because <type://ContentPreviewingClient> is evaluated first for the HTTP
 For <type://RetryingClient>, please check out 
 [`RetryingClient` with logging](/docs/client-retry#retryingclient-with-logging).
 
+## Modifying responses
+
+We can modify the responses in a decorator via wrapping the <type://HttpResponse> with <type://FilteredHttpResponse>:
+
+```java
+import com.linecorp.armeria.client.ClientBuilder;
+import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.common.FilteredHttpResponse;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpObject;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.ResponseHeaders;
+import java.nio.charset.StandardCharsets;
+
+ClientBuilder cb = Clients.builder(...);
+cb.decorator((delegate, ctx, req) -> {
+    HttpResponse response = delegate.execute(ctx, req);
+    return new FilteredHttpResponse(response) {
+        boolean initialHttpData = true;
+        protected HttpObject filter(HttpObject obj) {
+            if (obj instanceof ResponseHeaders) {
+                // Can receive multiple ResponseHeaders whose status is informational.
+                ResponseHeaders responseHeaders = (ResponseHeaders) obj;
+                if (responseHeaders.status().isInformational()) {
+                    return obj;
+                }
+                // Add a "x-foo" header into the response.
+                return responseHeaders.toBuilder()
+                                      .add("x-foo", "foobar")
+                                      .build();
+            } else if (obj instanceof HttpData) {
+                HttpData httpData = (HttpData) obj;
+                // Example of how to insert attribute into json payload.
+                // initial '{' in response is assumed to be first byte
+                if (initialHttpData) {
+                    initialHttpData = false;
+                    byte[] ascii = "{\"foo\": \"foobar\",".getBytes(StandardCharsets.US_ASCII);
+                    byte[] combined = Arrays.copyOf(ascii, ascii.length +  httpData.length() - 1);
+                    System.arraycopy(httpData.array(), 1, combined, ascii.length, httpData.length());
+                    return HttpData.wrap(combined);
+                }
+            } else {
+                // Http trailers.
+                assert obj instanceof HttpHeaders;
+            }
+            return obj;
+        }
+    };
+});
+```
+
+## Reacting to or logging responses
+
+If modification is not required, e.g. when handling quota responses from the service, and it doesn't need to look at the <type://HttpData>, it can be achieved in a more performant way via `ClientRequestContext.log().whenAvailable(RequestLogProperty).thenAccept(...)`. Accessing the <type://ClientRequestContext> can be done via either wrapping in a decorator (and using the ctx or by using <type://ClientRequestContextCaptor>:
```

Comment:
How about adding a reference to `/docs/advanced-structured-logging` in this section, like:

    See [Structured Logging](/docs/advanced-structued-logging) to learn more about the properties
    you can retrieve.

It'd also be nice if we wrap at 112th column like we do for our `.java` files.

Reply:
Thanks, and I've add that now. Yes, proper line breaks added. It's no excuse, but I wrote it in the Github UI, and didn't think about formatting.